File: 1-Iniciante/ex1042.c
#include <stdio.h>
#include <stdlib.h> 

void insertion(int n, int vet[]);
void exibeVet(int vet[], int n);
void copiaVet(int vet1[], int vet2[], int n);

int main(){
    //variaveis
    int *vet = malloc(3 * sizeof(int));
    int *vetOrdenado = malloc(3 * sizeof(int));

    //entrada
    scanf("%d %d %d", &vetOrdenado[0], &vetOrdenado[1], &vetOrdenado[2]);

    //processamento
    copiaVet(vetOrdenado, vet, 3);
    insertion(3, vetOrdenado);

    //saida
    exibeVet(vetOrdenado, 3);
    printf("\n");
    exibeVet(vet, 3);

    return 0;
}

void copiaVet(int vet1[], int vet2[], int n){
    int i;
    for(i = 0; i < n; i++){
        vet2[i] = vet1[i];
    }
}

void exibeVet(int vet[], int n){
    int i;
    for(i = 0; i < 3; i++){
        printf("%d\n", vet[i]);
    }
}

void insertion(int n, int vet[]){
    int i, j;
    int aux;
    for(i = 0; i < n; i++){
        aux = vet[i];
        for(j = i - 1; j >= 0 && vet[j] > aux; j--){
            vet[j + 1] = vet[j];
        }
        vet[j + 1] = aux;
    }
}

File: 1-Iniciante/Ex1042.java
import java.util.Scanner;
import java.util.Arrays;

public class Ex1042 { //para ser aceito no URI mude o nome da classe para Main
    public static void main(String[] args) {
        // variaveis
        Scanner scanner = new Scanner(System.in);
        String valores;
        String[] valoresSeparados;
        int[] vetOriginal = new int[3], vetOrdenado = new int[3];

        // entrada
        valores = scanner.nextLine();
        valoresSeparados = valores.split("\\s");

        // processamento
        for (int i = 0; i < valoresSeparados.length; i++) {
            vetOriginal[i] = Integer.parseInt(valoresSeparados[i]);
        }
        vetOrdenado = vetOriginal.clone();
        Arrays.sort(vetOrdenado);

        // saida
        imprimeVet(vetOrdenado, 3);
        System.out.println();
        imprimeVet(vetOriginal, 3);
        scanner.close();
    }

    public static void imprimeVet(int vet[], int n) {
        for (int i = 0; i < n; i++) {
            System.out.println(vet[i]);
        }
    }
}